source/gridfs/tests/delete.yml

```diff
@@ -1,123 +1,198 @@
-                    ] }
-    -
-        description: "Delete when files entry does not exist"
-        act:
-            operation: delete
-            arguments:
-                id: { "$oid" : "000000000000000000000000" }
-        assert:
-            error: "FileNotFound"
-    -
-        description: "Delete when files entry does not exist and there are orphaned chunks"
-        arrange:
-            data:
-                -
-                    { delete : "fs.files", deletes : [
-                        { q : { _id : { "$oid" : "000000000000000000000004" } }, limit : 1 }
-                    ] }
-        act:
-            operation: delete
-            arguments:
-                id: { "$oid" : "000000000000000000000004" }
-        assert:
-            error: "FileNotFound"
-            data:
-                -
-                    { delete : "expected.files", deletes : [
-                        { q : { _id : { "$oid" : "000000000000000000000004" } }, limit : 1 }
-                    ] }
+  - description: "delete when length is 0"
+    operations:
+      - name: delete
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000001" }
+    outcome:
+      - collectionName: *bucket0_files_collectionName
+        databaseName: *database0Name
+        documents:
+          - *file2
+          - *file3
+          - *file4
+      - collectionName: *bucket0_chunks_collectionName
+        databaseName: *database0Name
+        documents:
+          - *file2_chunk0
+          - *file3_chunk0
+          - *file4_chunk0
+          - *file4_chunk1
+  - description: "delete when length is 0 and there is one extra empty chunk"
+    operations:
+      - name: delete
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000002" }
+    outcome:
+      - collectionName: *bucket0_files_collectionName
+        databaseName: *database0Name
+        documents:
+          - *file1
+          - *file3
+          - *file4
+      - collectionName: *bucket0_chunks_collectionName
+        databaseName: *database0Name
+        documents:
+          - *file3_chunk0
+          - *file4_chunk0
+          - *file4_chunk1
+  - description: "delete when length is 8"
+    operations:
+      - name: delete
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000004" }
+    outcome:
+      - collectionName: *bucket0_files_collectionName
+        databaseName: *database0Name
+        documents:
+          - *file1
+          - *file2
+          - *file3
+      - collectionName: *bucket0_chunks_collectionName
+        databaseName: *database0Name
+        documents:
+          - *file2_chunk0
+          - *file3_chunk0
+  - description: "delete when files entry does not exist"
+    operations:
+      - name: delete
+        object: *bucket0
+        arguments:
+          id: { $oid: "000000000000000000000000" }
+        expectError: { isError: true } # FileNotFound
```

Comment:
The original tests asserted specific error types that were never actually mentioned in the spec. I've retained their names as comments to preserve historical context, as was done in the [POC tests](https://github.com/mongodb/specifications/commit/e4808b84f804f8b08293dcd51ce107e10a40589f#diff-45d7492ffd2b7db6b2eb70da128598f1f743076e786f1a2dcebec47408ae6bd5R87).

I also created [DRIVERS-1684](https://jira.mongodb.org/browse/DRIVERS-1684) should we decide to address this down the line.